spanner/api/Spanner.Samples.Tests/SpannerFixture.cs

```diff
@@ -0,0 +1,229 @@
+    public string ProjectId { get; } = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+    // Allow environment variables to override the default instance and database names.
+    public string InstanceId { get; } = Environment.GetEnvironmentVariable("TEST_SPANNER_INSTANCE") ?? "my-instance";
+    public string DatabaseId { get; } = Environment.GetEnvironmentVariable("TEST_SPANNER_DATABASE") ?? $"my-db-{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";
+    public string BackupDatabaseId { get; } = "my-test-database";
+    public string BackupId { get; } = "my-test-database-backup";
+    public string ToBeCancelledBackupId { get; } = $"my-backup-{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";
+    public string RestoredDatabaseId { get; } = $"my-restore-db-{DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()}";
+
+    public async Task InitializeAsync()
+    {
+        // Don't need to cleanup stale Backups and Databases when instance is new.
+        var isExistingInstance = InitializeInstance();
+        if (isExistingInstance)
+        {
+            await DeleteStaleBackupsAndDatabasesAsync();
+        }
+        await InitializeDatabaseAsync();
+        await InitializeBackupAsync();
+    }
+
+    public Task DisposeAsync()
+    {
+        return Task.CompletedTask;
+    }
+
+    private bool InitializeInstance()
+    {
+        InstanceAdminClient instanceAdminClient = InstanceAdminClient.Create();
+        InstanceName instanceName = InstanceName.FromProjectInstance(ProjectId, InstanceId);
+        try
+        {
+            Instance response = instanceAdminClient.GetInstance(instanceName);
+            return true;
+        }
+        catch (RpcException ex) when (ex.Status.StatusCode == StatusCode.NotFound)
+        {
+            CreateInstanceSample createInstanceSample = new CreateInstanceSample();
+            createInstanceSample.CreateInstance(ProjectId, InstanceId);
+            return false;
+        }
+    }
+
+    private async Task DeleteStaleBackupsAndDatabasesAsync()
+    {
+        DatabaseAdminClient databaseAdminClient = DatabaseAdminClient.Create();
+        var instanceName = InstanceName.FromProjectInstance(ProjectId, InstanceId);
+        var databases = databaseAdminClient.ListDatabases(instanceName)
+            .Where(c => c.DatabaseName.DatabaseId.StartsWith("my-db-") || c.DatabaseName.DatabaseId.StartsWith("my-restore-db-"));
+        var databasesToDelete = new List<string>();
+
+        // Delete all the databases created before 72 hrs.
+        var timestamp = DateTimeOffset.UtcNow.AddHours(-72).ToUnixTimeMilliseconds();
+        foreach (var database in databases)
+        {
+            var databaseId = database.DatabaseName.DatabaseId.Replace("my-restore-db-", "").Replace("my-db-", "");
+            if (long.TryParse(databaseId, out long dbCreationTime) && dbCreationTime <= timestamp)
+            {
+                databasesToDelete.Add(database.DatabaseName.DatabaseId);
+            }
+        }
```

Comment:
I want to insist on this: I know that `createTime` says "if exists", but are we certain that it's actually empty sometimes?
@skuruppu Can you confirm with the API team?

It would be so much cleaner to add that condition to the Where a few lines above instead of having to do this.

Reply:
Sorry for the delay here. I've sent a question to the API team and I'm awaiting a reply.

Reply:
I've got the answer. The `createTime` is currently not set on databases that weren't created as part of a RestoreDatabase request meaning that a database created with DatabaseCreate will not have this field set. But in an upcoming release, this field will be set for all databases.

I'm not sure what the timeline is but I think it should take about a month.

Reply:
OK then, we can revisit later.